qa/L0_logging/log_format_test.py

```diff
@@ -0,0 +1,271 @@
+            os.path.join(module_directory, "log_models")
+        )
+        self._server_process = None
+        self._server_options["log-file"] = os.path.join(
+            test_logs_directory, test_case_name + ".server.log"
+        )
+
+    def _launch_server(self, unescaped=None):
+        cmd = ["tritonserver"]
+
+        for key, value in self._server_options.items():
+            cmd.append(f"--{key}={value}")
+
+        env = os.environ.copy()
+
+        if unescaped:
+            env["TRITON_SERVER_ESCAPE_LOG_MESSSAGES"] = "FALSE"
+        elif unescaped is not None:
+            env["TRITON_SERVER_ESCAPE_LOG_MESSSAGES"] = "TRUE"
+
+        self._server_process = subprocess.Popen(
+            cmd,
+            env=env,
+            stdin=subprocess.DEVNULL,
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
+        )
+
+        wait_time = 5
+
+        while wait_time and not os.path.exists(self._server_options["log-file"]):
+            time.sleep(1)
+            wait_time -= 1
+
+        if not os.path.exists(self._server_options["log-file"]):
+            raise Exception("Log not found")
+
+    def validate_log_record(self, record, format_regex):
+        match = format_regex.search(record)
+        if match:
+            for field, value in match.groupdict().items():
+                if field in validators:
+                    try:
+                        validators[field](value)
+                    except Exception as e:
+                        raise type(e)(f"{e}\nInvalid {field}: '{match.group(field)}' in log record '{record}'")
+
+        else:
+            raise Exception("Invalid log line")
+
+    def verify_log_format(self, file_path, format_regex):
+        log_records = []
+        with open(file_path, "rt") as file_:
+            current_log_record = []
+            for line in file_:
+                match = format_regex.search(line)
+                if match:
+                    if current_log_record:
+                        log_records.append(current_log_record)
+                    current_log_record = [line]
+                else:
+                    current_log_record.append(line)
+        log_records.append(current_log_record)
+        log_records = ["".join(log_record_lines) for log_record_lines in log_records]
+        for log_record in log_records:
+            self.validate_log_record(log_record, format_regex)
+
+    @pytest.mark.parametrize(
+        "log_format,format_regex",
+        FORMATS,
+        ids=IDS,
+    )
+    def test_log_format(self, log_format, format_regex):
+        self._server_options["log-format"] = log_format.replace("_unescaped", "")
+        self._launch_server(unescaped=True if "_unescaped" in log_format else False)
+        time.sleep(1)
+        self._server_process.kill()
+        return_code = self._server_process.wait()
+        if isinstance(format_regex, str):
+            return
```

Comment:
I didn't understand this at a glance, but I'm assuming this is checking whether it's a regex or an empty string `""` from your definition of FORMATS at the top? 

Can it just be something like this then?
```python
if format_regex:
  self.verify_log_format(self._server_options["log-file"], format_regex)
```

And are we testing anything for the cases where `log_format` is not`"default"`?

Reply:
this is still place holder to test out the parameterization - will remove once have the regex for ISO in place